utils/request.py

```diff
@@ -0,0 +1,236 @@
+
+        return url, kwargs
+
+    def form_prime_path(self, endpoint: str) -> str:
+        """
+        Returns a url pointing at the requested endpoint for the Prime API.
+
+        :param endpoint: Endpoint to target, e.g. "/moves"
+        :return: full url to use in requests
+        """
+        if is_local(env=self.env):
+            base_domain = self.form_base_domain(
+                local_port="9443",
+                local_protocol="https",
+                local_subdomain="primelocal",
+            )
+        else:
+            base_domain = self.form_base_domain(deployed_subdomain="api")
+
+        return f"{base_domain}/prime/v1{endpoint}"
+
+    def prep_prime_request(self, endpoint: str) -> tuple[str, RequestKwargsType]:
+        """
+        Prepares a request URL and the request kwargs for making a request to the Prime API.
+        :param endpoint: endpoint to target, e.g. "/moves"
+        :return: tuple of the full url to the endpoint and the kwargs to pass to the request
+        """
+        url = self.form_prime_path(endpoint=endpoint)
+
+        kwargs = self.get_request_kwargs(certs_required=True)
+
+        return url, kwargs
+
+    def form_support_path(self, endpoint: str) -> str:
+        """
+        Returns a url pointing at the requested endpoint for the Support API.
+
+        :param endpoint: Endpoint to target, e.g. "/moves"
+        :return: full url to use in requests
+        """
+        if is_local(env=self.env):
+            base_domain = self.form_base_domain(
+                local_port="9443",
+                local_protocol="https",
+                local_subdomain="primelocal",
+            )
+        else:
+            base_domain = self.form_base_domain(deployed_subdomain="api")
+
+        return f"{base_domain}/support/v1{endpoint}"
+
+    def prep_support_request(self, endpoint: str) -> tuple[str, RequestKwargsType]:
+        """
+        Prepares a request URL and the request kwargs for making a request to the Support API.
+        :param endpoint: endpoint to target, e.g. "/moves"
+        :return: tuple of the full url to the endpoint and the kwargs to pass to the request
+        """
+        url = self.form_support_path(endpoint=endpoint)
+
+        kwargs = self.get_request_kwargs(certs_required=True)
+
+        return url, kwargs
+
+
+class MilMoveRequestMixin:
+    """
+    Mixin for a Locust TaskSet class/subclass that provides access to a helper for forming request
+    URLs and preparing request kwargs.
+    """
+
+    # The environment the host is running in (ex. locally or deployed).
+    # Can be any of the values in MilMoveEnv.
+    env: MilMoveEnv
+
+    request_preparer: MilMoveRequestPreparer
+
+    def __init__(self: Union[TaskSet, "MilMoveRequestMixin"], *args, **kwargs) -> None:
+        """
+        Sets up the env based on value from the --host flag. Note that the `self.host` value is set
+        on the class from the command line flag BEFORE initialization.
```

Comment:
This is right but could be overridden through the host form field if they start a new load test through the UI.

Reply:
Ah good point, I'll update this.